### ISSUE TITLE
Light Persistence

### ISSUE THREAD

**Issue by spripe** [icanos/hassio-plejd#294]
### Description

This PR addresses an issue in the hassio-plejd integration where dim levels and lighting states are inadvertently changed upon restarting the service. Currently, if a light (e.g., Light A) was on just before hassio-plejd was stopped, and the lighting is turned off via plejd during the downtime, upon restarting hassio-plejd, the light's state does not revert to the last known state as expected. Instead, the integration seems to set the lighting to a presumed state without consulting the actual last-known state.

### Expected Behavior

The ideal functionality would be for the integration not to make assumptions about the desired lighting state or dimming level upon restart. It should instead 'listen' to the current state and update the actual value via MQTT before setting any values actively.

Ignoring received MQTT messages while doing discovery seems to work:
https://github.com/tomjaimz/hassio-plejd/commit/f827806c362293c51aa8725f04e8a50470f63ffb

### Solution?

- Implement a check to fetch the current state from BLE for lights and dimmers before setting the state in the system over MQTT.
- Added a state retention feature to remember the last-known state before shutdown and use it as a reference(MQTT) upon restart if no updated state is received from BLE.
- Included additional logging to help troubleshoot and ensure the correct states are being set.

**Comment by SweVictor** [comment]
@spripe - what does "This PR" refer to? This is an issue and not a PR. Feel free to open a new PR _towards the develop branch_ if you have a suggested code fix in mind.

There is a PR #268 that might work, but I think it's at best a partial fix. 

There is also #260.

The issue has been discussed at length in #218.
I've added what I consider to be the best way forward in https://github.com/icanos/hassio-plejd/issues/218#issuecomment-1683747009. I started some work in this direction (MQTT mainly), but got derailed when a new HA update suddenly caused this addon to stop working (now resolved).
